cmake/main/CMakeLists.txt

```diff
@@ -0,0 +1,247 @@
+set(ATANI_DCC_BASEDIR ${COMPONENT_PATH}/../..)
+set(SYSLIB_DIR ${COMPONENT_PATH}/../syslib)
+
+set(SYSLIB_SRCS
+    ${SYSLIB_DIR}/AsyncTCP/src/AsyncTCP.cpp
+    ${SYSLIB_DIR}/ESPAsyncWebServer/src/AsyncEventSource.cpp
+    ${SYSLIB_DIR}/ESPAsyncWebServer/src/AsyncWebSocket.cpp
+    ${SYSLIB_DIR}/ESPAsyncWebServer/src/SPIFFSEditor.cpp
+    ${SYSLIB_DIR}/ESPAsyncWebServer/src/WebAuthentication.cpp
+    ${SYSLIB_DIR}/ESPAsyncWebServer/src/WebHandlers.cpp
+    ${SYSLIB_DIR}/ESPAsyncWebServer/src/WebRequest.cpp
+    ${SYSLIB_DIR}/ESPAsyncWebServer/src/WebResponses.cpp
+    ${SYSLIB_DIR}/ESPAsyncWebServer/src/WebServer.cpp
+    ${SYSLIB_DIR}/esp8266-oled-ssd1306/src/OLEDDisplay.cpp
+    ${SYSLIB_DIR}/esp8266-oled-ssd1306/src/OLEDDisplayUi.cpp
+    ${SYSLIB_DIR}/LiquidCrystal_PCF8574/src/LiquidCrystal_PCF8574.cpp
+)
+
+set(SYSLIB_INCS
+    ${SYSLIB_DIR}/ArduinoJson/src
+    ${SYSLIB_DIR}/AsyncTCP/src
+    ${SYSLIB_DIR}/esp8266-oled-ssd1306/src
+    ${SYSLIB_DIR}/LiquidCrystal_PCF8574/src
+    ${SYSLIB_DIR}/ESPAsyncWebServer/src
+    ${SYSLIB_DIR}/ArduinoJson
+    ${SYSLIB_DIR}/ArduinoJson/src/ArduinoJson
+    ${SYSLIB_DIR}/ArduinoJson/src/ArduinoJson/Data
+    ${SYSLIB_DIR}/ArduinoJson/src/ArduinoJson/Deserialization
+    ${SYSLIB_DIR}/ArduinoJson/src/ArduinoJson/Polyfills
+    ${SYSLIB_DIR}/ArduinoJson/src/ArduinoJson/Serialization
+    ${SYSLIB_DIR}/ArduinoJson/src/ArduinoJson/StringTraits
+    ${SYSLIB_DIR}/ArduinoJson/src/ArduinoJson/TypeTraits
+)
+
+
+
+
+
+set(DCCPPLIB_BASE_DIR ${ATANI_DCC_BASEDIR}/lib)
+set(DCCPPLIB_INCS
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/os
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/dcc
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/executor
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/freertos_drivers/esp32
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/freertos_drivers/arduino
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/openlcb
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src/sys
+    ${DCCPPLIB_BASE_DIR}/OpenMRNLite/src
+    ${DCCPPLIB_BASE_DIR}/NeoNextion/src
+    ${DCCPPLIB_BASE_DIR}/LocoNet2/src
```

Comment:
Split this by library to SRC/INC. is there a way to do wildcards in CMakeFile.txt?

Reply:
there are ways to do that, but in my experience specifying the individual files is the better approach. See here for example https://stackoverflow.com/questions/1027247/is-it-better-to-specify-source-files-with-glob-or-each-file-individually-in-cmak/18538444#18538444

Reply:
Ok, let's go with named files in that case